File: app/src/main/java/com/example/projectsantander/ui/LoginContract.java
package com.example.projectsantander.ui;

import android.content.Context;

public class LoginContract {


        interface View {
            void showProgress(final boolean show);

            void showError(String error);

            void navigateToHome();

            void navigateToHomeAdmin();

            Context getContext();
        }

        interface Presenter {
            void login(String username, String password);
        }
    }

File: bankapp/app/src/main/java/dev/ornelas/bankapp/login/LoginContract.kt
package dev.ornelas.bankapp.login

interface LoginContract {
    interface View {
        fun navigateToStatement(loginViewModel: LoginViewModel)
        fun displaySavedUser(username: String)
        fun setUserNameError()
        fun setUserPasswordError()
    }

    interface Presenter {
        fun onLogin(username: String, password: String)
    }
}